elastalert/ruletypes.py

```diff
@@ -58,9 +58,16 @@ def add_match(self, event):
         if ts in event:
             event[ts] = dt_to_ts(event[ts])
 
-        if self.rules.get('compound_query_key') and '_source' in event:
-            event.update({key: lookup_es_key(event['_source'], key) for key in self.rules['compound_query_key']})
-
+        query_key = self.rules.get('query_key')
+        event_key = event.get('key')
+        # make sure events have the query_key values in the query_key fields
+        # this is only relevant when we have a "key" field, which is Elastalert's
+        # way to provide information when lacking document data
+        if query_key is not None and event_key is not None:
+            split_query_key = [x.strip() for x in query_key.split(',')]
+            split_key = [x.strip() for x in event_key.split(',')]
```

Comment:
maybe for this it's just better to keep a `compound_key` field with arrays of values, to be added to the match wherever the actual key is added to it in a concatenated form?

Reply:
I tested and checked in the query_key fix. As for the value (key), the event is built individually by the different rules, so there doesn't seem to be a time in which converting the key to an array makes sense until this function or afterwards.